### ISSUE TITLE
Object Never Used (type <class 'tensorflow.python.framework.ops.Tensor'>):

### ISSUE THREAD

**Issue by jw447** [tensorflow/models#6252]
Please go to Stack Overflow for help and support:

http://stackoverflow.com/questions/tagged/tensorflow

Also, please understand that many of the models included in this repository are experimental and research-style code. If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **What is the top-level directory of the model you are using**: models/research/inception
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Only some simple printf to collect timing
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: CentOS
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: r1.12
- **Bazel version (if compiling from source)**: 0.15.0
- **CUDA/cuDNN version**: none
- **GPU model and memory**: none
- **Exact command to reproduce**:`bazel-bin/inception/imagenet_distributed_train --batch_size=32 --data_dir=/global/cscratch1/sd/jw447/imagenet-data --job_name='ps' --task_id=0 $ps_hosts $worker_hosts`

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

The distributed training of inception failed at the end of training. Both workers output the same error:

> ERROR:tensorflow:==================================
Object was never used (type <class 'tensorflow.python.framework.ops.Tensor'>):
<tf.Tensor 'report_uninitialized_variables/boolean_mask/GatherV2:0' shape=(?,) dtype=string>
If you want to mark it as used call its "mark_used()" method.
It was originally created here:
  File "/global/cscratch1/sd/jw447/models/research/inception/bazel-bin/inception/imagenet_distributed_train.runfiles/inception/inception/imagenet_distributed_train.py", line 69, in <module>
    tf.app.run()  File "/global/homes/j/jw447/.conda/envs/python-tf/lib/python3.6/site-packages/tensorflow/python/platform/app.py", line 125, in run
    _sys.exit(main(argv))  File "/global/cscratch1/sd/jw447/models/research/inception/bazel-bin/inception/imagenet_distributed_train.runfiles/inception/inception/imagenet_distributed_train.py", line 65, in main
    inception_distributed_train.train(server.target, dataset, cluster_spec)  File "/global/cscratch1/sd/jw447/models/research/inception/bazel-bin/inception/imagenet_distributed_train.runfiles/inception/inception/inception_distributed_train.py", line 329, in train
    tf.logging.info("Time for saving model checkpoint: %.3f sec" % duration)  File "/global/homes/j/jw447/.conda/envs/python-tf/lib/python3.6/site-packages/tensorflow/python/training/sync_replicas_optimizer.py", line 343, in apply_gradients
    return train_op  File "/global/homes/j/jw447/.conda/envs/python-tf/lib/python3.6/site-packages/tensorflow/python/util/tf_should_use.py", line 189, in wrapped
    return _add_should_use_warning(fn(*args, **kwargs))
==================================
INFO:tensorflow:Time for saving model checkpoint: 4.109 sec
ERROR:tensorflow:==================================
Object was never used (type <class 'tensorflow.python.framework.ops.Tensor'>):
<tf.Tensor 'report_uninitialized_variables/boolean_mask/GatherV2:0' shape=(?,) dtype=string>
If you want to mark it as used call its "mark_used()" method.
It was originally created here:
  File "/global/cscratch1/sd/jw447/models/research/inception/bazel-bin/inception/imagenet_distributed_train.runfiles/inception/inception/imagenet_distributed_train.py", line 69, in <module>
    tf.app.run()  File "/global/homes/j/jw447/.conda/envs/python-tf/lib/python3.6/site-packages/tensorflow/python/platform/app.py", line 125, in run
    _sys.exit(main(argv))  File "/global/cscratch1/sd/jw447/models/research/inception/bazel-bin/inception/imagenet_distributed_train.runfiles/inception/inception/imagenet_distributed_train.py", line 65, in main
    inception_distributed_train.train(server.target, dataset, cluster_spec)  File "/global/cscratch1/sd/jw447/models/research/inception/bazel-bin/inception/imagenet_distributed_train.runfiles/inception/inception/inception_distributed_train.py", line 329, in train
    tf.logging.info("Time for saving model checkpoint: %.3f sec" % duration)  File "/global/homes/j/jw447/.conda/envs/python-tf/lib/python3.6/site-packages/tensorflow/python/training/sync_replicas_optimizer.py", line 343, in apply_gradients
    return train_op  File "/global/homes/j/jw447/.conda/envs/python-tf/lib/python3.6/site-packages/tensorflow/python/util/tf_should_use.py", line 189, in wrapped
    return _add_should_use_warning(fn(*args, **kwargs))

The error occurred after the saver.save() function. 

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

Source code can be found from the models/research/inception.

**Comment by jw447** [comment]
I'm closing this issue now. The related issue and solution can be found at: #11823 

TL;DR: the SyncReplicaOptimizer is no longer supported by tensorflow. DistributedStrategy can be used to replace the functionalities.

Distributed Strategy: https://github.com/tensorflow/tensorflow/blob/r1.13/tensorflow/contrib/distribute/README.md

**Comment by shafiq-islam-cse** [comment]
The same thing happened with me... solution got from here thank you @jw447 .

**Comment by sirvnvu98** [comment]
@jw447  Can you give me details on how to fix this problem. I don't know how to fix it
